sql/updates/latest-dev.sql

```diff
@@ -128,13 +134,29 @@ ALTER TABLE _timescaledb_config.bgw_job ADD COLUMN config JSONB;
 ALTER TABLE _timescaledb_config.bgw_job DROP CONSTRAINT valid_job_type;
 ALTER TABLE _timescaledb_config.bgw_job ADD CONSTRAINT valid_job_type CHECK (job_type IN ('telemetry_and_version_check_if_enabled', 'reorder', 'drop_chunks', 'continuous_aggregate', 'compress_chunks', 'custom'));
 
+-- migrate reorder jobs
+UPDATE
+  _timescaledb_config.bgw_job job
+SET proc_name = 'policy_reorder',
+  proc_schema = '_timescaledb_internal',
+  config = jsonb_build_object('hypertable_id', reorder.hypertable_id, 'index_name', reorder.hypertable_index_name),
```

Comment:
Do we need the `hypertable_id` config if we also set the `hypertable_id` column in the jobs table? It seems to me that only the `index_name` is necessary.

Reply:
the hypertable_id in the jobs table is only used for dependency tracking, it makes sense to have all configuration values in the config object because that is part of the function call

Reply:
i had it not in the jsonb at first but then decided to add it because it makes the function call more generic

Reply:
I have the same comment: why do we need hypertable_id in the json? It is already part of the bgw_job table. I don't get what you mean by  "makes the function call generic " : what function are you referring to here?

Reply:
the function call signature for scheduled functions is (job_id int32, config jsonb ), having the hypertable_id in the config makes the function implementation independent from job because all required values are in config. So it's a tradeoff between non-redundant internal storage and cleaner implementation of external API.